website/source/docs/providers/vsphere/r/virtual_machine.html.markdown

```diff
@@ -113,6 +113,7 @@ The `disk` block supports:
 * `template` - (Required if size and bootable_vmdk_path not provided) Template for this disk.
 * `datastore` - (Optional) Datastore for this disk
 * `size` - (Required if template and bootable_vmdks_path not provided) Size of this disk (in GB).
+* `name` - (Required if size is provided) Name of the new disk file to create.
```

Comment:
Sounds good to me! I'll update this. I was operating on the assumption that size is only specified when we are creating a disk for the user.  Admittedly, this is not obvious to users so explicitly calling it out in the docs is smart.